### ISSUE TITLE
Edits to `oci_containerengine_cluster.endpoint_config.subnet_id` should trigger recreation

### ISSUE THREAD

**Issue by 12345ieee** [oracle/terraform-provider-oci#1834]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version and Provider Version

```
$ terraform -v
Terraform v1.4.4
on linux_amd64
[...]
+ provider registry.terraform.io/oracle/oci v4.114.0
```

### Affected Resource(s)

`oci_containerengine_cluster`

### Terraform Configuration Files

```hcl
resource "oci_containerengine_cluster" "k8s_cluster" {
  endpoint_config {
    is_public_ip_enabled = true
    nsg_ids              = "some ocid"
    # subnet_id            = "OLD ocid"
    subnet_id            = "NEW ocid"
  }
  
  [...]
}
```
### Actual Behavior

When changing the ocid of the control plane subnet at `oci_containerengine_cluster.endpoint_config.subnet_id` one gets a plan of this sort:
```diff
~ endpoint_config {
    ~ subnet_id            = "OLD" -> "NEW"
      # (2 unchanged attributes hidden)
  }
```
but this is not going to work, because as the documentation says, the control plane subnet is not updatable.

In fact after apply nothing changes and the same plan is proposed anew.

### Expected Behavior

Changing `oci_containerengine_cluster.endpoint_config.subnet_id` should trigger a recreation of the whole `oci_containerengine_cluster`.

### Steps to Reproduce

1. `terraform apply`
2. `terraform plan`
3. see nothing changed

**Comment by tf-oci-pub** [comment]
Thank you for reporting the issue. We have raised an internal ticket to track this. Our service engineers will get back to you.

**Comment by XinruXiao-9** [comment]
Hi, thanks for reporting this issue. 

OKE does not allow updating subnet_id in endpoint_config by design. Please check your terraform code and  stop changing this endpoint_config.subnet_id. 

We also understand this is causing confusion. We will add an error message when customer is trying to update endpoint_config.subnet_id. The ETA for the fix will be May 25.

Thanks!

**Comment by 12345ieee** [comment]
I am confused by your suggested fix.

I'd like if trying to change the endpoint subnet would propose a cluster destroy+create at plan time, like if I tried to change the subnet of an `oci_core_instance`.

You seem to be suggesting letting the change to go through at plan time and then popping an error at apply time, like what happens when you try to give a vnic a private ip that's already in use.

The second solution is much worse for the end user. Did I misunderstand something?

**Comment by Maxrovr** [comment]
Fixed in version 4.123.0 by @XinruXiao-9
